crates/ide_assists/src/handlers/move_guard.rs

```diff
@@ -149,6 +146,32 @@ pub(crate) fn move_arm_cond_to_match_guard(acc: &mut Assists, ctx: &AssistContex
             }
 
             edit.insert(match_pat.syntax().text_range().end(), buf);
+
+            // If with only an else branch
+            if let Some(ElseBranch::Block(else_block)) = if_expr.else_branch() {
+                let then_arm_end = match_arm.syntax().text_range().end();
+                if then_block.tail_expr().is_some() && then_only_expr {
+                    // Insert comma for expression if there isn't one
+                    match match_arm.syntax().last_child_or_token() {
+                        Some(NodeOrToken::Token(t)) if t.kind() == COMMA => {}
+                        _ => edit.insert(then_arm_end, ","),
+                    }
+                }
+                let else_only_expr = else_block.statements().next().is_none();
+                let indent_level = match_arm.indent_level();
+                let spaces = "    ".repeat(indent_level.0 as _);
+                edit.insert(then_arm_end, format!("\n{}{} => ", spaces, match_pat));
+                match &else_block.tail_expr() {
+                    Some(else_expr) if else_only_expr => {
+                        edit.insert(then_arm_end, else_expr.syntax().text());
+                        edit.insert(then_arm_end, ",");
+                    }
+                    _ if replace_node != *if_expr.syntax() => {
+                        edit.insert(then_arm_end, else_block.dedent(1.into()).syntax().text());
+                    }
+                    _ => edit.insert(then_arm_end, else_block.syntax().text()),
```

Comment:
can you add some `cov_mark::check!`s for the three branches here?

Basically adding a `cov_mark::hit!(move_guard_ifelse_expr_only);` to the first arm, `cov_mark::check!(move_guard_ifelse_expr_only);` to one of the tests that hits this match arm, same for the other two with a different identifier in the macro.

That way we can assert that those tests test these arms properly.